lib/persistence/dao/melee_weapon_dao.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:battle_it_out/persistence/dao/dao.dart';
+import 'package:battle_it_out/persistence/dao/item_quality_dao.dart';
+import 'package:battle_it_out/persistence/dao/skill_dao.dart';
+import 'package:battle_it_out/persistence/entities/attribute.dart';
+import 'package:battle_it_out/persistence/entities/item_quality.dart';
+import 'package:battle_it_out/persistence/entities/melee_weapon.dart';
+import 'package:battle_it_out/persistence/entities/skill.dart';
+import 'package:battle_it_out/persistence/database_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+class MeleeWeaponDAO extends DAO<MeleeWeapon> {
+  Map<int, Skill> skills;
+  Map<int, Attribute> attributes;
+
+  MeleeWeaponDAO(this.attributes, this.skills);
+
+  @override
+  get tableName => 'weapons_melee';
+
+  Future<List<ItemQuality>> getQualities(int id) async {
```

Comment:
I believe this method only differs from the one in [armour_dao.dart](https://github.com/BattleItOut/BattleItOut/pull/23/files#diff-841e1835b9b79f5cd1ffeefede7888ec90add53ea6aca4942f08fcc1e0185a01) by two static strings, but I guess that it won't be changing often and it's quite short, so it probably won't hurt us much.